extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```
@@ -0,0 +1,886 @@
+  const localAutofixPath = getAutofixPath();
+  if (!localAutofixPath) {
+    throw new Error(
+      "Path to local autofix installation not found. Internal GitHub access required.",
+    );
+  }
+  if (!(await pathExists(localAutofixPath))) {
+    throw new Error(`Local autofix path ${localAutofixPath} does not exist.`);
+  }
+  return localAutofixPath;
+}
+
+/**
+ * Overrides the query help from a given variant analysis
+ * at a location within the `localAutofixPath` directory .
+ */
+async function overrideQueryHelp(
+  variantAnalysis: VariantAnalysis,
+  cliServer: CodeQLCliServer,
+  localAutofixPath: string,
+): Promise<void> {
+  // Get path to the query used by the variant analysis.
+  const queryFilePath = variantAnalysis.query.filePath;
+  if (!(await pathExists(queryFilePath))) {
+    throw new Error(`Query file used by variant analysis not found.`);
+  }
+  const parsedQueryFilePath = parse(queryFilePath);
+  const queryFilePathNoExt = join(
+    parsedQueryFilePath.dir,
+    parsedQueryFilePath.name,
+  );
+
+  // Get the path to the query help, which may be either a `.qhelp` or a `.md` file.
+  // Note: we assume that the name of the query file is the same as the name of the query help file.
+  const queryHelpFilePathQhelp = `${queryFilePathNoExt}.qhelp`;
+  const queryHelpFilePathMarkdown = `${queryFilePathNoExt}.md`;
+
+  // Set `queryHelpFilePath` to the existing extension type.
+  let queryHelpFilePath: string;
+  if (await pathExists(queryHelpFilePathQhelp)) {
+    queryHelpFilePath = queryHelpFilePathQhelp;
+  } else if (await pathExists(queryHelpFilePathMarkdown)) {
+    queryHelpFilePath = queryHelpFilePathMarkdown;
+  } else {
+    throw new Error(
+      `Could not find query help file at either ${queryHelpFilePathQhelp} or ${queryHelpFilePathMarkdown}. Check that the query help file exists and is named correctly.`,
+    );
+  }
+
+  // Get the query metadata.
+  const metadata = await tryGetQueryMetadata(cliServer, queryFilePath);
```

Since the metadata is required, would it make sense to just call `cliServer.resolveMetadata` directly and let it throw an error if the metadata is not available? It seems like the `tryGetQueryMetadata` was created for cases where the query metadata is optional.